From 70f3283955786c63da5b5003986805f6b74ad5d4 Mon Sep 17 00:00:00 2001
From: Arun Sharma <asharma@fb.com>
Date: Tue, 26 Jul 2011 16:09:06 -0700
Subject: atomic: use <linux/atomic.h>

This allows us to move duplicated code in <asm/atomic.h>
(atomic_inc_not_zero() for now) to <linux/atomic.h>

Signed-off-by: Arun Sharma <asharma@fb.com>
Reviewed-by: Eric Dumazet <eric.dumazet@gmail.com>
Cc: Ingo Molnar <mingo@elte.hu>
Cc: David Miller <davem@davemloft.net>
Cc: Eric Dumazet <eric.dumazet@gmail.com>
Acked-by: Mike Frysinger <vapier@gentoo.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
(cherry picked from commit 60063497a95e716c9a689af3be2687d261f115b4)

Conflicts:

	arch/alpha/include/asm/atomic.h
	arch/alpha/include/asm/local.h
	arch/alpha/kernel/perf_event.c
	arch/alpha/kernel/smp.c
	arch/alpha/lib/dec_and_lock.c
	arch/arm/include/asm/atomic.h
	arch/arm/kernel/smp.c
	arch/arm/kernel/traps.c
	arch/arm/mach-at91/pm.c
	arch/arm/mach-bcmring/dma.c
	arch/arm/mach-cns3xxx/include/mach/pm.h
	arch/arm/mach-cns3xxx/pm.c
	arch/arm/mach-omap1/pm.c
	arch/arm/mach-s3c2440/clock.c
	arch/arm/mach-s3c2440/s3c2442.c
	arch/arm/mach-s3c2440/s3c244x-clock.c
	arch/avr32/include/asm/atomic.h
	arch/blackfin/include/asm/atomic.h
	arch/blackfin/include/asm/dma.h
	arch/blackfin/include/asm/ipipe.h
	arch/blackfin/include/asm/spinlock.h
	arch/blackfin/kernel/ftrace.c
	arch/blackfin/kernel/ipipe.c
	arch/blackfin/kernel/nmi.c
	arch/blackfin/mach-common/smp.c
	arch/cris/arch-v32/drivers/cryptocop.c
	arch/cris/arch-v32/kernel/smp.c
	arch/cris/include/asm/atomic.h
	arch/cris/include/asm/bitops.h
	arch/cris/kernel/process.c
	arch/frv/include/asm/atomic.h
	arch/frv/include/asm/hardirq.h
	arch/frv/kernel/irq.c
	arch/h8300/include/asm/atomic.h
	arch/ia64/include/asm/atomic.h
	arch/ia64/include/asm/processor.h
	arch/ia64/include/asm/spinlock.h
	arch/ia64/kernel/smp.c
	arch/ia64/kernel/smpboot.c
	arch/ia64/kernel/uncached.c
	arch/m32r/include/asm/atomic.h
	arch/m32r/include/asm/mmu_context.h
	arch/m32r/include/asm/spinlock.h
	arch/m32r/kernel/smp.c
	arch/m32r/kernel/traps.c
	arch/m68k/include/asm/atomic.h
	arch/microblaze/include/asm/mmu_context_mm.h
	arch/microblaze/include/asm/prom.h
	arch/mips/include/asm/atomic.h
	arch/mips/include/asm/hw_irq.h
	arch/mips/include/asm/local.h
	arch/mips/include/asm/smp.h
	arch/mips/kernel/irq.c
	arch/mips/kernel/mips-mt.c
	arch/mips/kernel/rtlx.c
	arch/mips/kernel/smp-cmp.c
	arch/mips/kernel/smp-mt.c
	arch/mips/kernel/smp.c
	arch/mips/kernel/smtc-proc.c
	arch/mips/kernel/smtc.c
	arch/mips/kernel/sync-r4k.c
	arch/mips/kernel/vpe.c
	arch/mips/mipssim/sim_smtc.c
	arch/mips/sgi-ip27/ip27-nmi.c
	arch/mn10300/include/asm/atomic.h
	arch/mn10300/include/asm/mmu_context.h
	arch/mn10300/include/asm/spinlock.h
	arch/mn10300/include/asm/system.h
	arch/mn10300/kernel/mn10300-watchdog.c
	arch/mn10300/kernel/traps.c
	arch/mn10300/mm/misalignment.c
	arch/mn10300/proc-mn2ws0050/proc-init.c
	arch/parisc/include/asm/atomic.h
	arch/parisc/include/asm/bitops.h
	arch/parisc/include/asm/mmu_context.h
	arch/parisc/kernel/parisc_ksyms.c
	arch/parisc/kernel/smp.c
	arch/parisc/kernel/traps.c
	arch/parisc/lib/bitops.c
	arch/powerpc/include/asm/atomic.h
	arch/powerpc/include/asm/emulated_ops.h
	arch/powerpc/include/asm/irq.h
	arch/powerpc/include/asm/local.h
	arch/powerpc/include/asm/prom.h
	arch/powerpc/kernel/of_platform.c
	arch/powerpc/kernel/ppc_ksyms.c
	arch/powerpc/kernel/rtas.c
	arch/powerpc/kernel/rtasd.c
	arch/powerpc/kernel/smp-tbsync.c
	arch/powerpc/kernel/smp.c
	arch/powerpc/platforms/83xx/km83xx.c
	arch/powerpc/platforms/83xx/mpc832x_mds.c
	arch/powerpc/platforms/83xx/mpc834x_itx.c
	arch/powerpc/platforms/83xx/mpc834x_mds.c
	arch/powerpc/platforms/83xx/mpc836x_mds.c
	arch/powerpc/platforms/83xx/sbc834x.c
	arch/powerpc/platforms/85xx/mpc85xx_cds.c
	arch/powerpc/platforms/85xx/mpc85xx_mds.c
	arch/powerpc/platforms/85xx/sbc8548.c
	arch/powerpc/platforms/cell/cpufreq_spudemand.c
	arch/powerpc/platforms/cell/smp.c
	arch/powerpc/platforms/cell/spufs/context.c
	arch/powerpc/platforms/chrp/smp.c
	arch/powerpc/platforms/iseries/smp.c
	arch/powerpc/platforms/powermac/backlight.c
	arch/powerpc/platforms/powermac/smp.c
	arch/powerpc/platforms/pseries/eeh.c
	arch/powerpc/platforms/pseries/eeh_cache.c
	arch/powerpc/platforms/pseries/smp.c
	arch/powerpc/sysdev/fsl_soc.c
	arch/powerpc/sysdev/tsi108_dev.c
	arch/s390/include/asm/atomic.h
	arch/s390/kernel/dis.c
	arch/s390/kernel/traps.c
	arch/sh/include/asm/atomic.h
	arch/sh/include/asm/hw_irq.h
	arch/sh/include/asm/smp.h
	arch/sh/kernel/idle.c
	arch/sh/kernel/smp.c
	arch/sh/kernel/traps_64.c
	arch/sh/kernel/unwinder.c
	arch/sparc/include/asm/atomic_32.h
	arch/sparc/include/asm/atomic_64.h
	arch/sparc/include/asm/prom.h
	arch/sparc/include/asm/smp_32.h
	arch/sparc/include/asm/smp_64.h
	arch/sparc/kernel/irq_64.c
	arch/sparc/kernel/leon_smp.c
	arch/sparc/kernel/perf_event.c
	arch/sparc/kernel/smp_32.c
	arch/sparc/kernel/smp_64.c
	arch/sparc/lib/atomic32.c
	arch/tile/include/asm/atomic.h
	arch/tile/include/asm/atomic_32.h
	arch/tile/include/asm/atomic_64.h
	arch/tile/include/asm/bitops_32.h
	arch/tile/include/asm/bitops_64.h
	arch/tile/include/asm/spinlock_32.h
	arch/tile/kernel/intvec_32.S
	arch/tile/lib/atomic_32.c
	arch/tile/lib/atomic_asm_32.S
	arch/x86/ia32/sys_ia32.c
	arch/x86/include/asm/apic.h
	arch/x86/include/asm/atomic.h
	arch/x86/include/asm/hw_irq.h
	arch/x86/include/asm/local.h
	arch/x86/include/asm/mce.h
	arch/x86/include/asm/mmu_context.h
	arch/x86/include/asm/prom.h
	arch/x86/include/asm/spinlock.h
	arch/x86/include/asm/thread_info.h
	arch/x86/kernel/amd_gart_64.c
	arch/x86/kernel/apic/apic.c
	arch/x86/kernel/apic/es7000_32.c
	arch/x86/kernel/cpu/common.c
	arch/x86/kernel/i8259.c
	arch/x86/kernel/irqinit.c
	arch/x86/kernel/traps.c
	arch/x86/kvm/lapic.c
	arch/x86/kvm/timer.c
	arch/x86/lib/atomic64_32.c
	arch/x86/mm/mmio-mod.c
	arch/xtensa/include/asm/atomic.h
	arch/xtensa/kernel/process.c
	crypto/af_alg.c
	crypto/proc.c
	crypto/rng.c
	drivers/atm/ambassador.c
	drivers/atm/atmtcp.c
	drivers/atm/eni.c
	drivers/atm/eni.h
	drivers/atm/firestream.c
	drivers/atm/fore200e.c
	drivers/atm/horizon.c
	drivers/atm/idt77252.c
	drivers/atm/iphase.c
	drivers/atm/nicstar.c
	drivers/atm/suni.c
	drivers/atm/uPD98402.c
	drivers/atm/zatm.c
	drivers/base/memory.c
	drivers/base/power/sysfs.c
	drivers/block/cciss_scsi.c
	drivers/char/ipmi/ipmi_watchdog.c
	drivers/char/mspec.c
	drivers/connector/cn_proc.c
	drivers/edac/edac_stub.c
	drivers/firewire/core-card.c
	drivers/firewire/core-device.c
	drivers/firewire/core-topology.c
	drivers/firewire/core.h
	drivers/firewire/nosy.c
	drivers/gpu/drm/radeon/radeon.h
	drivers/gpu/drm/radeon/radeon_fence.c
	drivers/gpu/drm/ttm/ttm_bo.c
	drivers/gpu/drm/ttm/ttm_lock.c
	drivers/gpu/drm/ttm/ttm_object.c
	drivers/gpu/drm/ttm/ttm_page_alloc.c
	drivers/hwmon/sht15.c
	drivers/infiniband/hw/cxgb4/mem.c
	drivers/infiniband/hw/ehca/ehca_tools.h
	drivers/infiniband/hw/nes/nes_cm.c
	drivers/infiniband/ulp/ipoib/ipoib.h
	drivers/infiniband/ulp/srp/ib_srp.c
	drivers/isdn/gigaset/gigaset.h
	drivers/md/dm-crypt.c
	drivers/md/dm-kcopyd.c
	drivers/md/dm-mpath.c
	drivers/md/dm-queue-length.c
	drivers/md/dm-table.c
	drivers/media/video/hdpvr/hdpvr-core.c
	drivers/media/video/tlg2300/pd-dvb.c
	drivers/media/video/uvc/uvc_ctrl.c
	drivers/media/video/uvc/uvc_queue.c
	drivers/media/video/uvc/uvc_v4l2.c
	drivers/media/video/uvc/uvc_video.c
	drivers/message/i2o/i2o_scsi.c
	drivers/misc/phantom.c
	drivers/net/atlx/atl1.c
	drivers/net/atlx/atl2.c
	drivers/net/atlx/atl2.h
	drivers/net/cassini.c
	drivers/net/cpmac.c
	drivers/net/cxgb3/cxgb3_offload.c
	drivers/net/cxgb3/l2t.h
	drivers/net/cxgb3/t3cdev.h
	drivers/net/cxgb4/cxgb4_uld.h
	drivers/net/cxgb4/l2t.h
	drivers/net/hamradio/6pack.c
	drivers/net/hamradio/dmascc.c
	drivers/net/ibmveth.c
	drivers/net/phy/phy.c
	drivers/net/ppp_generic.c
	drivers/net/wimax/i2400m/i2400m.h
	drivers/net/wireless/b43legacy/b43legacy.h
	drivers/net/wireless/b43legacy/dma.h
	drivers/oprofile/oprofile_stats.h
	drivers/pci/hotplug/cpci_hotplug_core.c
	drivers/pci/xen-pcifront.c
	drivers/s390/block/dasd_eer.c
	drivers/s390/char/sclp_quiesce.c
	drivers/s390/char/vmlogrdr.c
	drivers/s390/cio/device.h
	drivers/s390/cio/qdio_main.c
	drivers/s390/cio/qdio_thinint.c
	drivers/s390/crypto/ap_bus.c
	drivers/s390/crypto/zcrypt_api.c
	drivers/s390/crypto/zcrypt_cex2a.c
	drivers/s390/crypto/zcrypt_mono.c
	drivers/s390/crypto/zcrypt_pcica.c
	drivers/s390/crypto/zcrypt_pcicc.c
	drivers/s390/crypto/zcrypt_pcixcc.c
	drivers/s390/net/fsm.h
	drivers/s390/scsi/zfcp_scsi.c
	drivers/sbus/char/display7seg.c
	drivers/scsi/dpt/dpti_i2o.h
	drivers/scsi/hpsa.c
	drivers/scsi/pm8001/pm8001_sas.h
	drivers/staging/octeon/ethernet-rx.c
	drivers/staging/octeon/ethernet-tx.c
	drivers/staging/solo6x10/solo6x10.h
	drivers/staging/tidspbridge/include/dspbridge/host_os.h
	drivers/staging/winbond/mds_s.h
	drivers/staging/winbond/wb35reg_s.h
	drivers/tty/bfin_jtag_comm.c
	drivers/tty/rocket.c
	drivers/tty/serial/dz.c
	drivers/tty/serial/sb1250-duart.c
	drivers/tty/serial/zs.c
	drivers/usb/gadget/f_audio.c
	drivers/usb/gadget/f_rndis.c
	drivers/usb/gadget/uvc_queue.c
	drivers/usb/image/microtek.c
	drivers/usb/misc/appledisplay.c
	drivers/usb/serial/garmin_gps.c
	drivers/usb/wusbcore/wa-rpipe.c
	drivers/vhost/vhost.h
	drivers/video/vermilion/vermilion.h
	drivers/w1/masters/matrox_w1.c
	drivers/w1/w1.c
	drivers/w1/w1_family.h
	drivers/watchdog/intel_scu_watchdog.c
	drivers/watchdog/sbc7240_wdt.c
	fs/btrfs/delayed-inode.h
	fs/direct-io.c
	fs/eventpoll.c
	fs/file_table.c
	fs/gfs2/main.c
	fs/nfs/cache_lib.h
	fs/nfs/direct.c
	fs/notify/group.c
	fs/notify/inode_mark.c
	fs/notify/mark.c
	fs/notify/notification.c
	fs/notify/vfsmount_mark.c
	fs/ntfs/inode.h
	fs/posix_acl.c
	fs/proc/meminfo.c
	include/acpi/platform/aclinux.h
	include/asm-generic/atomic.h
	include/asm-generic/local.h
	include/asm-generic/local64.h
	include/drm/ttm/ttm_lock.h
	include/linux/aio.h
	include/linux/atmdev.h
	include/linux/atomic.h
	include/linux/backing-dev.h
	include/linux/bit_spinlock.h
	include/linux/buffer_head.h
	include/linux/configfs.h
	include/linux/connector.h
	include/linux/cred.h
	include/linux/crypto.h
	include/linux/dcache.h
	include/linux/debug_locks.h
	include/linux/device.h
	include/linux/edac.h
	include/linux/fault-inject.h
	include/linux/fdtable.h
	include/linux/filter.h
	include/linux/firewire.h
	include/linux/fsnotify_backend.h
	include/linux/interrupt.h
	include/linux/jump_label.h
	include/linux/kdb.h
	include/linux/key.h
	include/linux/kgdb.h
	include/linux/kobject.h
	include/linux/mlx4/device.h
	include/linux/mman.h
	include/linux/mmzone.h
	include/linux/mount.h
	include/linux/mutex.h
	include/linux/netdevice.h
	include/linux/nfs_fs_sb.h
	include/linux/oprofile.h
	include/linux/pci.h
	include/linux/perf_event.h
	include/linux/phy.h
	include/linux/proc_fs.h
	include/linux/quota.h
	include/linux/rwsem.h
	include/linux/sem.h
	include/linux/skbuff.h
	include/linux/sonet.h
	include/linux/spinlock.h
	include/linux/sunrpc/auth.h
	include/linux/sunrpc/cache.h
	include/linux/sunrpc/timer.h
	include/linux/swap.h
	include/linux/sysfs.h
	include/linux/vmstat.h
	include/linux/workqueue.h
	include/net/ax25.h
	include/net/cipso_ipv4.h
	include/net/flow.h
	include/net/inet_hashtables.h
	include/net/inet_timewait_sock.h
	include/net/inetpeer.h
	include/net/ip_vs.h
	include/net/lib80211.h
	include/net/llc.h
	include/net/neighbour.h
	include/net/net_namespace.h
	include/net/netfilter/nf_conntrack.h
	include/net/netlabel.h
	include/net/netns/conntrack.h
	include/net/sctp/structs.h
	include/pcmcia/ds.h
	include/rdma/ib_sa.h
	include/rdma/ib_verbs.h
	include/rxrpc/types.h
	include/scsi/scsi_device.h
	kernel/audit.c
	kernel/auditsc.c
	kernel/cgroup.c
	kernel/cpuset.c
	kernel/debug/debug_core.c
	kernel/rcupdate.c
	kernel/rcutorture.c
	kernel/rcutree_trace.c
	kernel/rwsem.c
	kernel/stop_machine.c
	kernel/taskstats.c
	kernel/trace/trace.h
	kernel/trace/trace_mmiotrace.c
	lib/atomic64.c
	lib/atomic64_test.c
	lib/crc32.c
	lib/dec_and_lock.c
	mm/init-mm.c
	mm/kmemleak.c
	mm/slob.c
	mm/vmalloc.c
	net/atm/atm_misc.c
	net/atm/clip.c
	net/atm/common.c
	net/atm/lec.c
	net/atm/proc.c
	net/bridge/br_fdb.c
	net/core/flow.c
	net/decnet/dn_fib.c
	net/decnet/dn_neigh.c
	net/decnet/dn_table.c
	net/decnet/dn_timer.c
	net/ipv4/cipso_ipv4.c
	net/ipv4/raw.c
	net/ipv6/ip6_tunnel.c
	net/iucv/iucv.c
	net/l2tp/l2tp_core.c
	net/l2tp/l2tp_ppp.c
	net/netfilter/nfnetlink_log.c
	net/netfilter/nfnetlink_queue.c
	net/netlabel/netlabel_cipso_v4.c
	net/netlabel/netlabel_kapi.c
	net/netlabel/netlabel_mgmt.c
	net/netlabel/netlabel_mgmt.h
	net/netlabel/netlabel_unlabeled.c
	net/sunrpc/xprtrdma/xprt_rdma.h
	net/tipc/core.h
	security/selinux/hooks.c
	security/selinux/xfrm.c
	sound/pci/echoaudio/darla20.c
	sound/pci/echoaudio/darla24.c
	sound/pci/echoaudio/echo3g.c
	sound/pci/echoaudio/gina20.c
	sound/pci/echoaudio/gina24.c
	sound/pci/echoaudio/indigo.c
	sound/pci/echoaudio/indigodj.c
	sound/pci/echoaudio/indigodjx.c
	sound/pci/echoaudio/indigoio.c
	sound/pci/echoaudio/indigoiox.c
	sound/pci/echoaudio/layla20.c
	sound/pci/echoaudio/layla24.c
	sound/pci/echoaudio/mia.c
	sound/pci/echoaudio/mona.c
	sound/pci/lx6464es/lx6464es.h
	sound/sparc/dbri.c

Signed-off-by: Simon Horman <horms@verge.net.au>
---
 drivers/video/sh_mobile_lcdcfb.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/sh_mobile_lcdcfb.c b/drivers/video/sh_mobile_lcdcfb.c
index 019dbd3..b048417 100644
--- a/drivers/video/sh_mobile_lcdcfb.c
+++ b/drivers/video/sh_mobile_lcdcfb.c
@@ -24,7 +24,7 @@
 #include <linux/backlight.h>
 #include <linux/gpio.h>
 #include <video/sh_mobile_lcdc.h>
-#include <asm/atomic.h>
+#include <linux/atomic.h>
 
 #include "sh_mobile_lcdcfb.h"
 #include "sh_mobile_meram.h"
-- 
1.7.10

